### ISSUE TITLE
Rule proposal: no-named-export

### ISSUE THREAD

**Issue by fsmaia** [benmosher/eslint-plugin-import#1156]
Although it has uncommon usages, it would be nice to have a `no-named-export` rule to enforce export default only.

This may be useful, for example, when defining that a component file may not expose its helpers, giving this file a single purpose, forcing helpers to be put in a proper file/layer with specific rules and behaviors; and organizing those module exports in an index file.

In a practical example:
```js
// valid
import React from 'react';

const App = () => (<div className="App">App</div>);

export default App;
```

```js
// valid
import React from 'react';

const title = 'App';

const App = () => (<div className="App">{title}</div>);

export default App;
```

```js
// invalid
import React from 'react';

export const title = 'App';

const App = () => (<div className="App">{title}</div>);

export default App;
```
